pkg/remote/aws/repository/ecr_repository_test.go

```diff
@@ -88,3 +91,73 @@ func Test_ecrRepository_ListAllRepositories(t *testing.T) {
 		})
 	}
 }
+
+func Test_ecrRepository_GetRepositoryPolicy(t *testing.T) {
+	dummyError := errors.New("this is an error")
+
+	tests := []struct {
+		name    string
+		mocks   func(client *awstest.MockFakeECR)
+		want    *ecr.GetRepositoryPolicyOutput
+		wantErr error
+	}{
+		{
+			name: "Get repository policy",
+			mocks: func(client *awstest.MockFakeECR) {
+				client.On("GetRepositoryPolicy",
+					&ecr.GetRepositoryPolicyInput{
+						RegistryId:     aws.String("1"),
+						RepositoryName: aws.String("2"),
+					},
+				).Return(&ecr.GetRepositoryPolicyOutput{
+					RegistryId:     aws.String("1"),
+					RepositoryName: aws.String("2"),
+				}, nil).Once()
+			},
+			want: &ecr.GetRepositoryPolicyOutput{
+				RegistryId:     aws.String("1"),
+				RepositoryName: aws.String("2"),
+			},
+		},
+		{
+			name: "Get repository policy error",
+			mocks: func(client *awstest.MockFakeECR) {
+				client.On("GetRepositoryPolicy",
+					&ecr.GetRepositoryPolicyInput{
+						RegistryId:     aws.String("1"),
+						RepositoryName: aws.String("2"),
+					},
+				).Return(nil,
+					remoteerr.NewResourceListingError(dummyError, resourceaws.AwsEcrRepositoryPolicyResourceType),
```

Comment:
The repository will never return such errors, this can be confusing. Can you simply use `dummyError` as is ?